test/IDE/complete_non_default_literal.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: %batch-code-completion
+
+extension Double {
+  var doubleOnlyMember: Double { self }
+}
+
+func test() {
+  let double: Double = 2.#^DOUBLE_CONTEXTUAL_TYPE^#
+  // DOUBLE_CONTEXTUAL_TYPE: Decl[InstanceVar]/CurrNominal/TypeRelation[Convertible]: doubleOnlyMember[#Double#];
+
+  let int: Int = 2.#^INT_CONTEXTUAL_TYPE^#
+  // INT_CONTEXTUAL_TYPE: Decl[InstanceVar]/CurrNominal: doubleOnlyMember[#Double#];
```

Comment:
Could you add a test with custom `ExpressibleByIntegerLiteral` type?
```swift
struct CustomInt: ExpressibleByIntegerLiteral {
  public init() {}
  public init(integerLiteral value: Int) {  }
  var customIntMember -> CustomInt { .init() }
}

let foo: CustomInt = 2.#^COMPLETE^#
```
This should _not_ suggest `customIntMember` because it doesn't type check.